cmd/vulcan-vulners/main.go

```diff
@@ -259,20 +293,39 @@ func analyzeReport(target string, nmapReport *gonmap.NmapRun) ([]report.Vulnerab
 				continue
 			}
 
-			v, err := vulnerabilitiesByProdVers(port.Service.Product, port.Service.Version, "software")
+			f, err := findingByProdVers(port.Service.Product, port.Service.Version, "software")
 			if err != nil {
 				return nil, err
 			}
-			if v != nil {
-				v.Summary = fmt.Sprintf(v.Summary, port.Service.Product)
-				v.Description = fmt.Sprintf(v.Description, port.Service.Product)
-				v.Details = fmt.Sprintf("Found at %d", port.PortId)
-
-				vulns = append(vulns, *v)
+			summary := fmt.Sprintf(vulnersVuln.Summary, port.Service.Product)
+			v, ok := uniqueVulns[summary]
+			if !ok {
+				v.Vuln = report.Vulnerability{
+					Summary:         summary,
+					Description:     fmt.Sprintf(vulnersVuln.Description, port.Service.Product),
+					Score:           f.Score,
+					Recommendations: vulnersVuln.Recommendations,
+				}
+				v.CPEs = map[string]struct{}{}
+				uniqueVulns[summary] = v
 			}
+			productID := port.Service.Product + port.Service.Version
+			if _, ok := v.Products[productID]; !ok {
+				v.CPEs[productID] = struct{}{}
+				v.Vuln.Resources = append(v.Vuln.Resources, f.Resource)
+				uniqueVulns[summary] = v
+				if f.Score > v.Vuln.Score {
```

Comment:
Same as the previous comment regarding the score calculation. I think it's possible to miss some scores here.

Reply:
Yes well the assumption is the same, the score must be the same.

Reply:
ok!